reactor-core/src/main/java/reactor/core/publisher/Sinks.java

```diff
@@ -0,0 +1,262 @@
+	 *     pushed to the sink AFTER this subscriber was subscribed.</li>
+	 *     <li>Without {@link Subscriber}: Discarding. Pushing elements while there are no {@link Subscriber}
+	 *     registered will simply discard these elements instead of "warming up" the sink.</li>
+	 * </ul>
+	 */
+	@SuppressWarnings("deprecation")
+	public static <T> StandaloneFluxSink<T> multicastNoWarmup() {
+		return new FluxProcessorSink<>(ReplayProcessor.create(0));
+	}
+
+	/**
+	 * A {@link StandaloneFluxSink} with the following characteristics:
+	 * <ul>
+	 *     <li>Multicast</li>
+	 *     <li>Backpressure : this sink honors downstream demand of individual subscribers.</li>
+	 *     <li>Replaying: by {@code historySize}. Keeps the last {@code historySize} elements and
+	 *     replays them instantly to new subscribers before continuing with "live" elements.</li>
+	 *     <li>Without {@link Subscriber}: buffers enough elements pushed without a subscriber to
+	 *     honor the {@code historySize}.</li>
+	 * </ul>
+	 */
+	@SuppressWarnings("deprecation")
+	public static <T> StandaloneFluxSink<T> replay(int historySize) {
+		return new FluxProcessorSink<>(ReplayProcessor.create(historySize));
+	}
+
+	/**
+	 * A {@link StandaloneFluxSink} with the following characteristics:
+	 * <ul>
+	 *     <li>Multicast</li>
+	 *     <li>Backpressure : this sink honors downstream demand of individual subscribers.</li>
+	 *     <li>Replaying: all elements pushed to this sink are replayed to new subscribers.</li>
+	 *     <li>Without {@link Subscriber}: all elements pushed to this sink are remembered,
+	 *     even when there is no subscriber.</li>
+	 * </ul>
+	 */
+	@SuppressWarnings("deprecation")
+	public static <T> StandaloneFluxSink<T> replayAll() {
+		return new FluxProcessorSink<>(ReplayProcessor.create());
+	}
+
+	/**
+	 * A {@link StandaloneFluxSink} with the following characteristics:
+	 * <ul>
+	 *     <li><strong>Unicast</strong>: contrary to most other {@link StandaloneFluxSink}, the
+	 *     {@link Flux} view rejects {@link Subscriber subscribers} past the first one.</li>
+	 *     <li>Backpressure : this sink honors downstream demand of its single {@link Subscriber}.</li>
+	 *     <li>Replaying: non-applicable, since only one {@link Subscriber} can register.</li>
+	 *     <li>Without {@link Subscriber}: all elements pushed to this sink are remembered and will
+	 *     be replayed once the {@link Subscriber} subscribes.</li>
+	 * </ul>
+	 */
+	@SuppressWarnings("deprecation")
+	public static <T> StandaloneFluxSink<T> unicast() {
+		return new FluxProcessorSink<>(UnicastProcessor.create());
+	}
+
+	/**
+	 * A {@link StandaloneMonoSink} that can be triggered at any time by either of
+	 * the three completions: {@link ScalarSink#success(Object) valued completion},
+	 * {@link ScalarSink#success() empty completions} or {@link ScalarSink#error(Throwable) error}.
+	 * This completion is replayed to late subscribers.
+	 */
+	public static <T> StandaloneMonoSink<T> trigger() {
```

Comment:
WDYT about `promise` or `future`? `trigger` is too verb-ish

Reply:
I don't really like `trigger` either, future and promise sound too overloaded. Maybe `completable` to echo both `CompletableFuture` and `Completable` in RxJava? By I fear this becomes overloaded too...

Reply:
IIRC `Completable` in RxJava is `Publisher<Void>`. It does sound too overloaded but we're talking about method's name, not type, so _I think_ we're fine as it helps understanding the return value much better.

Reply:
ok, I'll go with `promise` actually...